### PR TITLE
modify signmessage payload schema nonce type

### PR DESCRIPTION
**Pull request by AmazingBeerBelly**
# Description

Modify SignMessage PayloadSchema's nonce type from [32] to ["u8"]

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

# Type of change.

- [x] FIX - a PR of this type patches a bug.
- [ ] FEATURE - a PR of this type introduces a new feature.
- [ ] BUILD - a PR of this type introduces build changes.
- [ ] CI - a PR of this type introduces CI changes.
- [ ] DOCS - a PR of this type introduces DOCS improvement.
- [ ] STYLE - a PR of this type introduces style changes.
- [ ] REFACTOR - a PR of this type introduces refactoring.
- [ ] PERFORMANCE - a PR of this type introduces performance changes.
- [ ] TEST - a PR of this type adds more tests.
- [ ] CHORE - a PR introduces other changes than the specified above.

### CODE REVIEW

File: packages/core/src/lib/helpers/verify-signature/payload.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const payloadSchema = new Map([
       fields: [
         ["tag", "u32"],
         ["message", "string"],
-        ["nonce", [32]],
+        ["nonce", ["u8"]],
```
Comment (kujtimprenkuSQA) [inline packages/core/src/lib/helpers/verify-signature/payload.ts:32]: Why is this needed? - this will break all other implementations.

Comment (kujtimprenkuSQA) [reply packages/core/src/lib/helpers/verify-signature/payload.ts:32]: The NEP413 states: 
>A nonce that uniquely identifies this instance of the message, denoted as a 32 bytes array (a fixed `Buffer` in JS/TS).

From my understanding in borsh-js `v0.7.0` an array of unsigned 8-bit integers with a fixed length of 32 elements is done as we already have `["nonce", [32]]`

By, changing the type to `["nonce", ["u8"]]` would allow any length of 8-bit integer elements.

Comment (AmazingBeerBelly) [reply packages/core/src/lib/helpers/verify-signature/payload.ts:32]: Ok, it's my fault. I using v1.0.0, and set struct to "nonce: { array: { type: 'u8', len: 32 } }" will work.

Close this PR.

Comment (kujtimprenkuSQA) [reply packages/core/src/lib/helpers/verify-signature/payload.ts:32]: I see, in v1.0.0 borsh has been re-written, you probably need to do something like this:

```ts
const schema = {
    struct: {
        tag: 'u32',
        message: 'string',
        nonce: { array: { type: 'u8', len: 32 } },
        recipient: 'string',
        callbackUrl: { option: 'string' },
    },
};
```
